flair/datasets/sequence_labeling.py

```diff
@@ -4973,6 +4973,209 @@ def _write_instances(cls, version, base_path, split, data):
                 out_file.write("\n")
 
 
+class NER_NOISEBENCH(ColumnCorpus):
+    label_url = "https://raw.githubusercontent.com/elenamer/NoiseBench/main/data/annotations/"
+    SAVE_TRAINDEV_FILE = False
+
+    def __init__(
+        self,
+        noise: str = "clean",
+        base_path: Optional[Union[str, Path]] = None,
+        in_memory: bool = True,
+        **corpusargs,
+    ) -> None:
+        """Initialize the NoiseBench corpus.
+
+        Args:
+            noise (string): Chooses the labelset for the data.
+                clean (default): Clean labels
+                crowd,crowdbest,expert,distant,weak,llm : Different kinds of noisy labelsets (details: ...)
+            base_path (Optional[Union[str, Path]]): Path to the data.
+                Default is None, meaning the corpus gets automatically downloaded and saved.
+                You can override this by passing a path to a directory containing the unprocessed files but typically this
+                should not be necessary.
+            in_memory (bool): If True the dataset is kept in memory achieving speedups in training.
+            **corpusargs: The arguments propagated to :meth:'flair.datasets.ColumnCorpus.__init__'.
+        """
+        if noise not in ["clean", "crowd", "crowdbest", "expert", "distant", "weak", "llm"]:
+            raise Exception("Please choose a valid version")
```

Comment:
It is recommended to raise [ValueError](https://docs.python.org/3/library/exceptions.html#:~:text=Passing%20arguments%20of%20the%20wrong%20type%20(e.g.%20passing%20a%20list%20when%20an%20int%20is%20expected)%20should%20result%20in%20a%20TypeError%2C%20but%20passing%20arguments%20with%20the%20wrong%20value%20(e.g.%20a%20number%20outside%20expected%20boundaries)%20should%20result%20in%20a%20ValueError.) for wrong and unexpected values passed to the argument.

Also, an error message would be more helpful if it listed valid values. Something like:
```python
if noise not in VALID_NOISE_VALUES:
    raise ValueError(f"Unsupported value for noise argument. Got {noise}, expected one of {VALUE_NOISE_VALUES}!")
```